rwengine/src/engine/Garage.cpp

```diff
@@ -0,0 +1,889 @@
+#include "Garage.hpp"
+
+#include <btBulletDynamicsCommon.h>
+#include <glm/gtx/quaternion.hpp>
+
+#include "dynamics/CollisionInstance.hpp"
+
+#include "ai/PlayerController.hpp"
+
+#include "engine/GameState.hpp"
+
+#include "objects/CharacterObject.hpp"
+#include "objects/GameObject.hpp"
+#include "objects/InstanceObject.hpp"
+#include "objects/VehicleObject.hpp"
+
+Garage::Garage(GameWorld* engine_, const int id_, const glm::vec3 coord0,
+               const glm::vec3 coord1, const Type type_)
+    : engine(engine_), id(id_), type(type_) {
+    min.x = std::min(coord0.x, coord1.x);
+    min.y = std::min(coord0.y, coord1.y);
+    min.z = std::min(coord0.z, coord1.z);
+
+    max.x = std::max(coord0.x, coord1.x);
+    max.y = std::max(coord0.y, coord1.y);
+    max.z = std::max(coord0.z, coord1.z);
+
+    glm::vec2 midpoint;
```

Comment:
It can also be const `glm::vec2 midpoint{(min.x + max.x) / 2, (min.y + max.y) / 2};`